applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp

```diff
@@ -24,1468 +24,1261 @@
-            
-            Values.SetStrainVector(this_constitutive_variables.StrainVector);
-            
-            // Reading integration points
-            const GeometryType::IntegrationPointsArrayType& integration_points = GetGeometry().IntegrationPoints(  );
+            if ( rOutput[point_number].size() != strain_size )
+                rOutput[point_number].resize( strain_size, false );
 
-            // Displacements vector
-            Vector displacements;
-            GetValuesVector(displacements);
-            
-            //reading integration points
-            for ( unsigned int point_number = 0; point_number < integration_points.size(); point_number++ )
-            {
-                // Compute element kinematics B, F, DN_DX ...
-                CalculateKinematicVariables(this_kinematic_variables, point_number, integration_points);
-
-                // Compute material reponse
-                CalculateConstitutiveVariables(this_kinematic_variables, this_constitutive_variables, Values, point_number, integration_points, GetStressMeasure(), displacements);
+            rOutput[point_number] = this_constitutive_variables.StressVector;
+        }
+    } else if( rVariable == GREEN_LAGRANGE_STRAIN_VECTOR  || rVariable == ALMANSI_STRAIN_VECTOR ) {
+        // Create and initialize element variables:
+        const unsigned int number_of_nodes = GetGeometry().size();
+        const unsigned int dimension = GetGeometry().WorkingSpaceDimension();
+        const unsigned int strain_size = mConstitutiveLawVector[0]->GetStrainSize();
 
-                if ( rOutput[point_number].size() != strain_size)
-                {
-                    rOutput[point_number].resize( strain_size, false );
-                }
+        KinematicVariables this_kinematic_variables(strain_size, dimension, number_of_nodes);
+        ConstitutiveVariables this_constitutive_variables(strain_size);
+        
+        // Create constitutive law parameters:
+        ConstitutiveLaw::Parameters Values(GetGeometry(),GetProperties(),rCurrentProcessInfo);
+        
+        // Set constitutive law flags:
+        Flags &ConstitutiveLawOptions=Values.GetOptions();
+        ConstitutiveLawOptions.Set(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN, true);
+        ConstitutiveLawOptions.Set(ConstitutiveLaw::COMPUTE_STRESS, false);
+        ConstitutiveLawOptions.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, false);
+        
+        Values.SetStrainVector(this_constitutive_variables.StrainVector);
+        
+        // Reading integration points
+        const GeometryType::IntegrationPointsArrayType& integration_points = GetGeometry().IntegrationPoints(  );
 
-                rOutput[point_number] = this_constitutive_variables.StrainVector;
-            }
-        }
-        else
-        {
-            if ( rOutput.size() != GetGeometry().IntegrationPoints(  ).size() )
-            {
-                rOutput.resize( GetGeometry().Integr
```

Comment:
👍 for moving this to the baseclass and using the new macros

Reply:
👍